### PR TITLE
Remove pl module and pytorch-lightning dependency.

### PR DESCRIPTION
**Pull request by MylesBartlett**

### CODE REVIEW

**Comment by thomkeh** [comment]
You have to edit `ranzen.rst` as well.

**Comment by thomkeh** [comment]
No, sorry, `index.rst`

**Comment by MylesBartlett** [comment]
> You have to edit `ranzen.rst` as well.

Yes, I've done that -- waiting for poetry to add colorama to the 'hydra' dependencies

**Comment by thomkeh** [comment]
Another thing you may already have done locally is removing this line from pyproject.toml:

```
pl​ = [​"​pytorch-lightning​"​, ​"​tqdm​"​]
```

File: pyproject.toml [hunk 1]
```diff
@@ -64,9 +63,8 @@ wandb = {version= "^0.12.0", optional = true}
 tqdm = {version = "^4.62.0", optional = true}
 
 [tool.poetry.extras]
-all = ["hydra-core", "neoconfigen", "pandas", "pytorch-lightning", "wandb"]
+all = ["hydra-core", "neoconfigen", "pandas", "wandb", "torch", "numpy"]
```
Comment (thomkeh) [inline pyproject.toml:66]: pretty sure `torch` depends on `numpy`

Comment (MylesBartlett) [reply pyproject.toml:66]: It does but there's no harm in being explicit

**Comment by thomkeh** [comment]
Some warnings from sphinx during building of the docs:
```
/home/runner/work/ranzen/ranzen/ranzen/torch/loss.py:docstring of ranzen.torch.loss.CrossEntropyLoss:1:<autosummary>:1: ERROR: Unknown interpreted text role "function".

/home/runner/work/ranzen/ranzen/ranzen/torch/__init__.py:docstring of ranzen.torch.sampling.batched_randint:1: ERROR: Unknown interpreted text role "function".
/home/runner/work/ranzen/ranzen/ranzen/torch/__init__.py:docstring of ranzen.torch.sampling.batched_randint:5: ERROR: Unknown interpreted text role "function".
```

**Pull request merged**